src/pool/srv_target.c

```diff
@@ -525,6 +525,17 @@ pool_fetch_hdls_ult(void *data)
 static void
 tgt_ec_eph_query_ult(void *data)
 {
+	struct ds_pool	*pool = data;
+
+	/* sp_map == NULL means the IV ns is not setup yet, i.e.
+	 * the pool leader does not broadcast the pool map to the
+	 * current node yet, see pool_iv_pre_sync().
+	 */
+	ABT_mutex_lock(pool->sp_mutex);
+	if (pool->sp_map == NULL)
+		ABT_cond_wait(pool->sp_map_cond, pool->sp_mutex);
```

Comment:
What if the ult doesn't wait here when pool->sp_map == NULL? I just don't quite see why it has to wait here.

Reply:
sp_map_iv == NULL means the pool leader has not broadcast the pool map to this node yet, i.e. the IV NS is not setup yet. If the target is excluded from the pool map, the leader will never send the pool map to this node until it is reintegrated again.

So if we do not wait here, and  an excluded target is being restarted (note: ds_pool service will still start in this case), then it will keep trying to do IV operation for EC epoch, and generate some annoying error message.

Reply:
so why don't we change the code to avoid the annoying error messages by checking "pool->sp_map"?

Reply:
may need to check "if (pool->sp_map == NULL)" again after wait? as in ds_pool_stop() it will signal that cond.
But looks not a big issue...